src/MultiMapManager/SafeParent.js

```diff
@@ -0,0 +1,54 @@
+import React from 'react'
+import ugh from 'ugh'
+
+export default class SafeParent extends React.Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      parentContextKey: null
+    }
+
+    this.ref = React.createRef()
+  }
+
+  componentDidMount () {
+    const { Component, providerProps } = this.props
+    const keys = Object.keys(providerProps)
+
+    if (this.ref.current) {
+      const parentContextKey = keys.find(key => {
+        const parentMap = this.ref.current.closest(`#${key}`)
+
+        return parentMap?.id
+      })
+
+      if (!parentContextKey) ugh.error(`Could not find parent <Map> for ${Component.name} during context lookup`)
```

Comment:
add warning to use the map div as the portal root

a portal root above the map parent will break closest